### PR TITLE
Allow android.support{v7,v14} for @PreferenceXXX

### LINKED ISSUE #1535: @PreferenceXXX should work with support-v7 PreferenceFragmentCompat
An [official implementation](https://plus.google.com/+AndroidDevelopers/posts/9kZ3SsXdT2T) just added for `PreferenceFragment` compatibility in the Support Library. Our annotations should support that. See https://github.com/excilys/androidannotations/issues/1396#issuecomment-135756979.

### PR DESCRIPTION
**Pull request by WonderCsabo**
Fixes #1535.

### CODE REVIEW

File: AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/AbstractPreferenceListenerHandler.java [hunk 1]
```diff
@@ -48,8 +48,8 @@ public void validate(Element element, ElementValidation valid) {
 	}
 
 	@Override
-	protected final JClass getListenerTargetClass() {
-		return getClasses().PREFERENCE;
+	protected final JClass getListenerTargetClass(HasPreferences holder) {
+		return holder.usingSupportV7Preference() ? getClasses().SUPPORT_V7_PREFERENCE : getClasses().PREFERENCE;
```
Comment (yDelouis) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/AbstractPreferenceListenerHandler.java]: This code : 
```java
holder.usingSupportV7Preference() ? getClasses().SUPPORT_V7_PREFERENCE : getClasses().PREFERENCE;
```
is repeated at least three times in the code. Maybe we could add a method `JClass getPreferenceClass()` in `HasPreferences`.

File: AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/android/support/v7/preference/PreferenceFragmentCompat.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ */
+package android.support.v7.preference;
+
+public class PreferenceFragmentCompat extends android.app.Fragment {
+
+	public void addPreferencesFromResource(int preferencesResId) {
+
+	}
+
+	public android.support.v7.preference.Preference findPreference(java.lang.CharSequence key) {
```
Comment (yDelouis) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/android/support/v7/preference/PreferenceFragmentCompat.java]: Fully qualified name is not need here, is it ?

**Comment by yDelouis** [comment]
It is good, but has some duplicated code.

File: AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/android/support/v7/preference/PreferenceFragmentCompat.java (hunk 2 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations-core/androidannotations/src/test/java/android/support/v7/preference/PreferenceFragmentCompat.java]: You are right. I quickly copied this from the decompiled source, that's why FQCNs are here.

**Comment by WonderCsabo** [comment]
@yDelouis updated.

**Comment by yDelouis** [comment]
Thanks !

**Pull request merged**